source/Options.cpp

```diff
@@ -147,336 +147,176 @@ Options::Options(
+            /* check_exist */ false);
+    }
 
-  generator_configuration_paths_ = parse_paths_list(
-      variables["model-generator-configuration-paths"].as<std::string>(),
-      /* extension */ ".json");
-  model_generators_configuration_ =
-      parse_json_configuration_files(generator_configuration_paths_);
+    if (!json["grepo-metadata-path"].isNull()) {
+        grepo_metadata_path_ =
+            check_path_exists(json["grepo-metadata-path"].asString());
+    }
 
-  if (!variables["model-generator-search-paths"].empty()) {
-    model_generator_search_paths_ = parse_search_paths(
-        variables["model-generator-search-paths"].as<std::string>());
-  }
+    apk_path_ = check_path_exists(json["apk-path"].asString());
+    output_directory_ = std::filesystem::path(
+        check_directory_exists(json["output-directory"].asString()));
 
-  repository_root_directory_ = check_directory_exists(
-      variables["repository-root-directory"].as<std::string>());
-  source_root_directory_ = check_directory_exists(
-      variables["source-root-directory"].as<std::string>());
+    if (!json["sharded-models-directory"].isNull()) {
+        sharded_models_directory_ = std::filesystem::path(check_directory_exists(
+            json["sharded-models-directory"].asString()));
+    }
 
-  if (!variables["source-exclude-directories"].empty()) {
-    source_exclude_directories_ = parse_paths_list(
-        variables["source-exclude-directories"].as<std::string>(),
-        /* extension */ std::nullopt,
-        /* check_exist */ false);
-  }
+    sequential_ = json["sequential"].asBool();
+    skip_source_indexing_ = json["skip-source-indexing"].asBool();
+    skip_analysis_ = json["skip-analysis"].asBool();
+    remove_unreachable_code_ = json["remove-unreachable-code"].asBool();
+    disable_parameter_type_overrides_ =
+        json["disable-parameter-type-overrides"].asBool();
+    disable_global_type_analysis_ = json["disable-global-type-analysis"].asBool();
+    verify_expected_output_ = json["verify-expected-output"].asBool();
+    maximum_method_analysis_time_ =
+        json["maximum-method-analysis-time"] ?
+        std::nullopt
+        : std::make_optional<int>(json["maximum-method-analysis-time"].asInt());
+    maximum_source_sink_distance_ =
+        json["maximum-source-sink-distance"].asInt();
+    emit_all_via_cast_features_ = 
+        json["emit-all-via-cast-features"].asBool();
+
+    if (!json["allow-via-cast-feature"].isNull()) {
+        for (const auto& value : json["allow-via-cast-feature"]) {
+            allow_via_cast_features_.push_back(value.asString());
+        }
+    }
 
-  if (!variables["grepo-metadata-path"].empty()) {
-    grepo_metadata_path_ =
-        check_path_exists(variables["grepo-metadata-path"].as<std::string>());
-  }
+    if (!json["log-method"].isNull()) {
+        for (const auto& value : json["log-method"]) {
+            log_methods_.push_back(value.asString());
+        }
+    }
```

Comment:
* Instead of `json[key].isNull()` prefer using `json.isMember(key)`
* This will crash if `json["log-method"]` is not an array. Use `JsonValidation::nonempty_array()`
* Use `JsonValidation::string` instead of `.asString()` because `.asString()` throws exceptions that we don't handle, but `JsonValidation::string` throws proper exceptions that we catch and provide to the user.

```suggestion
    if (json.isMember("log-method")) {
        for (const auto& value : JsonValidation::nonempty_array(json, "log-method")) {
            log_methods_.push_back(JsonValidation::string(value));
        }
    }
```